engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java

```diff
@@ -0,0 +1,155 @@
+package org.enso.compiler.pass.analyse;
+
+import java.io.IOException;
+import org.enso.compiler.core.Persistance;
+import org.enso.compiler.pass.analyse.AliasAnalysis.Graph;
+import org.enso.compiler.pass.resolve.DocumentationComments;
+import org.enso.compiler.pass.resolve.FullyQualifiedNames;
+import org.enso.compiler.pass.resolve.IgnoredBindings;
+import org.enso.compiler.pass.resolve.ModuleAnnotations;
+import org.enso.compiler.pass.resolve.TypeSignatures;
+import org.enso.interpreter.dsl.Persistable;
+import org.openide.util.lookup.ServiceProvider;
+import scala.Option;
+
+@Persistable(clazz = CachePreferenceAnalysis.WeightInfo.class, id = 1111)
+@Persistable(clazz = DataflowAnalysis.DependencyInfo.class, id = 1112)
+@Persistable(clazz = DataflowAnalysis.DependencyMapping.class, id = 1113)
+@Persistable(clazz = GatherDiagnostics.DiagnosticsMeta.class, id = 1114)
+@Persistable(clazz = DocumentationComments.Doc.class, id = 1115)
+@Persistable(clazz = AliasAnalysis$Info$Occurrence.class, id = 1116)
+@Persistable(clazz = TypeSignatures.Signature.class, id = 1117)
+@Persistable(clazz = ModuleAnnotations.Annotations.class, id = 1118)
+@Persistable(clazz = AliasAnalysis$Info$Scope$Root.class, id = 1120)
+@Persistable(clazz = DataflowAnalysis$DependencyInfo$Type$Static.class, id = 1121)
+@Persistable(clazz = DataflowAnalysis$DependencyInfo$Type$Dynamic.class, id = 1122)
+@Persistable(clazz = AliasAnalysis$Info$Scope$Child.class, id = 1123)
+@Persistable(clazz = AliasAnalysis$Graph$Occurrence$Use.class, id = 1125)
+@Persistable(clazz = AliasAnalysis$Graph$Occurrence$Def.class, id = 1126)
+@Persistable(clazz = AliasAnalysis$Graph$Link.class, id = 1127)
+@Persistable(clazz = FullyQualifiedNames.FQNResolution.class, id = 1128)
+@Persistable(clazz = FullyQualifiedNames.ResolvedLibrary.class, id = 1129)
+@Persistable(clazz = FullyQualifiedNames.ResolvedModule.class, id = 1130)
+public final class PassPersistance {
```

Comment:
If I understand correctly, this is really not a compiler pass, but rather serverves as a definition how to persist the classes in `@Persistable` annotation. Maybe put it in different package? Or rename it?

Reply:
Well, it is "persistence for passes". In short: `PassPersistance`. Renaming isn't not a problem.